### DISCUSSION TITLE
Passing tags parameter

### DISCUSSION THREAD

**Discussion by RobBowman** [actions/virtual-environments#7939] [General questions] [answered]
Does anyone have an example of calling the "GenerateResourcesAndImage" function and passing tags to be used when creating the azure resources please? The problem I'm facing is I'm getting back an error from Azure policy like this:

_policyDefinition":{"name":"Require a tag on resource groups_

This is what I've tried:

```
$tagsJson = '{"BusinessUnit":"IDS", "ProductName":"DevOpsAgent", "Environment":"Production"}'
$tagsHashtable = ($tagsJson | ConvertFrom-Json -AsHashtable)
GenerateResourcesAndImage -SubscriptionId sub-id-here -ResourceGroupName rg-name-here -AzureLocation "UK South" -ImageType "UbuntuMinimal" -Tags $tagsHashtable
```

It gives the following error:
```
on <value for var.azure_tags from arguments> line 1:
  (source code not available)

Variables may not be used here.

Error: Variables not allowed
```

**Comment by vpolikarpov-akvelon** [comment]
Hi @RobBowman. I tried to reproduce your issue, but didn't succeed. Did you make any changes to scripts or templates before trying to run this?

**Reply by tdysko-cf** [reply]
I got the same result no modifications done on scripts

**Reply by vpolikarpov-akvelon** [reply]
Hey @tdysko-cf. Could you provide exact commands you executed and Powershell version?

**Reply by tdysko-cf** [reply]
hi @vpolikarpov-akvelon sorry for the delay, simply clone the repo run `GenerateResourcesAndImage -Tags @{tag_one= "test"; environment= "test"; application_name = "test"}` just pick the ubuntu2204 or ubuntu minimal. Something is wrong with HCL. 

I was only able to pass that tags variable through an environment variable like this:
`$env:PKR_VAR_azure_tags = '{ tag_one = "test", environment = "qa", application_name = "vmss"}'` 
Passing it through CLI variables, like it's done in GenerateResourcesAndImage failed.

**Reply by vpolikarpov-akvelon** [reply, marked as answer]
Hey @tdysko-cf. I finally realised what causes this issue. I've fixed it in #8058. Could you please update your fork or local repo from main branch and try again?

**Reply by tdysko-cf** [reply]
looks good now, thanks!

**Comment by vpolikarpov-akvelon** [comment]
Hey @RobBowman @tdysko-cf.
JFYI: We rewrote image generation script recently (https://github.com/actions/runner-images/pull/8065). We implemented a lot of improvements and migrated from Az Powershell module to CLI tool. It should be backwards compatible. Feel free to reach out if you have any problems with the new version.

**Reply by RobBowman** [reply]
Thanks @vpolikarpov-akvelon it is working ok now
